File: python-sdk/src/boxsdk/models/shared/postoauth2token.py
import dataclasses
from typing import Optional
from enum import Enum

class PostOAuth2TokenActorTokenTypeEnum(str, Enum):
    URN_IETF_PARAMS_OAUTH_TOKEN_TYPE_ID_TOKEN = "urn:ietf:params:oauth:token-type:id_token"

class PostOAuth2TokenBoxSubjectTypeEnum(str, Enum):
    ENTERPRISE = "enterprise"
    USER = "user"

class PostOAuth2TokenGrantTypeEnum(str, Enum):
    AUTHORIZATION_CODE = "authorization_code"
    REFRESH_TOKEN = "refresh_token"
    CLIENT_CREDENTIALS = "client_credentials"
    URN_IETF_PARAMS_OAUTH_GRANT_TYPE_JWT_BEARER = "urn:ietf:params:oauth:grant-type:jwt-bearer"
    URN_IETF_PARAMS_OAUTH_GRANT_TYPE_TOKEN_EXCHANGE = "urn:ietf:params:oauth:grant-type:token-exchange"

class PostOAuth2TokenSubjectTokenTypeEnum(str, Enum):
    URN_IETF_PARAMS_OAUTH_TOKEN_TYPE_ACCESS_TOKEN = "urn:ietf:params:oauth:token-type:access_token"


@dataclasses.dataclass
class PostOAuth2Token:
    r"""PostOAuth2Token
    A request for a new OAuth 2.0 token
    """
    
    grant_type: PostOAuth2TokenGrantTypeEnum = dataclasses.field(metadata={'form': { 'field_name': 'grant_type' }})
    actor_token: Optional[str] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'actor_token' }})
    actor_token_type: Optional[PostOAuth2TokenActorTokenTypeEnum] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'actor_token_type' }})
    assertion: Optional[str] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'assertion' }})
    box_shared_link: Optional[str] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'box_shared_link' }})
    box_subject_id: Optional[str] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'box_subject_id' }})
    box_subject_type: Optional[PostOAuth2TokenBoxSubjectTypeEnum] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'box_subject_type' }})
    client_id: Optional[str] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'client_id' }})
    client_secret: Optional[str] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'client_secret' }})
    code: Optional[str] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'code' }})
    refresh_token: Optional[str] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'refresh_token' }})
    resource: Optional[str] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'resource' }})
    scope: Optional[str] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'scope' }})
    subject_token: Optional[str] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'subject_token' }})
    subject_token_type: Optional[PostOAuth2TokenSubjectTokenTypeEnum] = dataclasses.field(default=None, metadata={'form': { 'field_name': 'subject_token_type' }})
    

File: ts-sdk/src/sdk/models/shared/postoauth2token.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum PostOAuth2TokenActorTokenTypeEnum {
    UrnIetfParamsOauthTokenTypeIdToken = "urn:ietf:params:oauth:token-type:id_token"
}

export enum PostOAuth2TokenBoxSubjectTypeEnum {
    Enterprise = "enterprise",
    User = "user"
}

export enum PostOAuth2TokenGrantTypeEnum {
    AuthorizationCode = "authorization_code",
    RefreshToken = "refresh_token",
    ClientCredentials = "client_credentials",
    UrnIetfParamsOauthGrantTypeJwtBearer = "urn:ietf:params:oauth:grant-type:jwt-bearer",
    UrnIetfParamsOauthGrantTypeTokenExchange = "urn:ietf:params:oauth:grant-type:token-exchange"
}

export enum PostOAuth2TokenSubjectTokenTypeEnum {
    UrnIetfParamsOauthTokenTypeAccessToken = "urn:ietf:params:oauth:token-type:access_token"
}


// PostOAuth2Token
/** 
 * A request for a new OAuth 2.0 token
**/
export class PostOAuth2Token extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "form, name=actor_token;" })
  actorToken?: string;

  @SpeakeasyMetadata({ data: "form, name=actor_token_type;" })
  actorTokenType?: PostOAuth2TokenActorTokenTypeEnum;

  @SpeakeasyMetadata({ data: "form, name=assertion;" })
  assertion?: string;

  @SpeakeasyMetadata({ data: "form, name=box_shared_link;" })
  boxSharedLink?: string;

  @SpeakeasyMetadata({ data: "form, name=box_subject_id;" })
  boxSubjectId?: string;

  @SpeakeasyMetadata({ data: "form, name=box_subject_type;" })
  boxSubjectType?: PostOAuth2TokenBoxSubjectTypeEnum;

  @SpeakeasyMetadata({ data: "form, name=client_id;" })
  clientId?: string;

  @SpeakeasyMetadata({ data: "form, name=client_secret;" })
  clientSecret?: string;

  @SpeakeasyMetadata({ data: "form, name=code;" })
  code?: string;

  @SpeakeasyMetadata({ data: "form, name=grant_type;" })
  grantType: PostOAuth2TokenGrantTypeEnum;

  @SpeakeasyMetadata({ data: "form, name=refresh_token;" })
  refreshToken?: string;

  @SpeakeasyMetadata({ data: "form, name=resource;" })
  resource?: string;

  @SpeakeasyMetadata({ data: "form, name=scope;" })
  scope?: string;

  @SpeakeasyMetadata({ data: "form, name=subject_token;" })
  subjectToken?: string;

  @SpeakeasyMetadata({ data: "form, name=subject_token_type;" })
  subjectTokenType?: PostOAuth2TokenSubjectTokenTypeEnum;
}
